src/LoadBalancer.ts

```diff
@@ -0,0 +1,127 @@
+import { KDPoint, KDTree, Logger } from 'edumeet-common';
+import MediaNode from './media/MediaNode';
+import { Peer } from './Peer';
+import Room from './Room';
+import * as geoip from 'geoip-lite';
+
+const logger = new Logger('LoadBalancer');
+
+export interface LoadBalancerOptions {
+	kdTree: KDTree;
+	defaultClientPosition: KDPoint;
+	cpuLoadThreshold?: number;
+	geoDistanceThreshold?: number;
+}
+
+export default class LoadBalancer {
+	private readonly cpuLoadThreshold: number;
+	private readonly geoDistanceThreshold: number;
+	private kdTree: KDTree;
+	private readonly defaultClientPosition: KDPoint;
+
+	constructor({
+		kdTree,
+		defaultClientPosition,
+		cpuLoadThreshold = 0.85,
+		geoDistanceThreshold = 2000
+	}: LoadBalancerOptions) {
+		logger.debug('constructor() [kdTree: %s, defaultClientPosition: %S]', kdTree, defaultClientPosition);
+		this.kdTree = kdTree;
+		this.defaultClientPosition = defaultClientPosition;
+		this.cpuLoadThreshold = cpuLoadThreshold;
+		this.geoDistanceThreshold = geoDistanceThreshold;
+	}
+
+	public getCandidates(
+		room: Room,
+		peer: Peer
+	): MediaNode[] {
+		try {
+			logger.debug('getCandidates() [room.id: %s, peer.id: %s]', room.id, peer.id);
+			let candidates: MediaNode[] = [];
```

Comment:
This is a wasted assignment, will be re-assigned below

Reply:
Done